Ability to Hide Sets

Daynar100 opened the pull request:
I used branch ed6ff0a as a starting point and tried to follow the suggestions in my last pull request for finishing implementation. It should hide item sets when the '-' on the title bar is clicked. If the filter option is turned on it should show them all, and any sets that would otherwise be hidden should have a '+' which can be clicked to make them visible when the filter is unchecked. 

I did add id's to all item sets. (Hopefully I didn't miss any.) Edit: I did miss 2 and failed to add id to types, which is why I originally failed one of the tests. This has now been fixed. 
The id's were generally of the form SECTION_NAME (ie WARMIND_TRIALS as the test one was in the base hidden-sets branch.)

I wasn't able to get saving working like was suggested. I wasn't sure how to get the Inventory component to have access to the variable containing the hidden sets information. But I was able to get it working by calling an id specific save function along with the action in ItemSet. This works in basically the same way as saving did in my last pull request. I can probably get it working the other way, but I thought I should submit this first version to make sure everything else looks okay.

joshhunt commented:
This is awesome! I'll take a bit of a closer look tomorrow, but I'm loving this! Well done!

joshhunt commented on src/components/ItemSet/index.js:
```diff
@@ -80,3 +97,18 @@ export default function ItemSet({ className, setPopper, setModal, set }) {
+      itemDefs: selectedItemDefsSelector(state, ownProps)
+    };
+  };
+};
```
I don't think any of this `mapStateToProps` function is required any more? It was probably there when you started this work (and/or when the hide-sets branch was made), but its no longer needed.


I think that argument is optional, so you could just remove it entirely (have only `connect(mapDispatchToActions)`). Otherwise, just inline an function that returns empty object (`connect(() => ({}), mapDispatchToActions)`)

joshhunt commented on src/views/Inventory/selectors.js:
```diff
@@ -142,16 +144,27 @@ const setDataSelector = createSelector(
-  (filters, setData, inventory, itemDefs) => {
+  (filters, hiddenSets, setData, inventory, itemDefs) => {
     const prevWhitelistedItems = ls.getTempFilterItemWhitelist();
+    //const hiddenSets = ['WARMIND_TRIALS'];
```
can you remove this line?

joshhunt commented on src/types.js:
```diff
@@ -18,6 +18,7 @@ export type SetSection = {|
 
 export type DestinySet = {|
   name: string,
+  id: string,
```
oh perfect 👌

joshhunt commented on src/components/ItemSet/index.js:
```diff
@@ -1,13 +1,23 @@
+import * as ls from 'app/lib/ls';
```
could you move this down to above `import Item from 'app/components/Item';`? I keep all external dependencies first, then our app-level imports.

Daynar100 commented:
I removed the test comment, moved the import, and removed mapStateToProps. I looked up connect [here](https://github.com/reduxjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) and it looks like mapDispatchToActions needs to be the second argument, with null being suggested if mapStateToProps isn't needed. If you think an inline function would be better I can do that instead.

joshhunt commented:
I might make a few interface tweaks to this after it goes it, but I think this is good for now! Congrats on your first open source contribution 🎉

joshhunt commented:
hey @Daynar100 do you have a twitter? Would love to give you a bit of a shoutout

Daynar100 commented:
No, I don't really use social media. Thank you though!
